dsl/camel-jbang/camel-jbang-plugin-k/src/main/java/org/apache/camel/dsl/jbang/core/commands/k/Agent.java

```diff
@@ -0,0 +1,424 @@
+                server.close();
+            }
+            if (vertx != null) {
+                vertx.close();
+            }
+        }
+
+        return 0;
+    }
+
+    // Visible for testing
+    CompletionStage<HttpServer> serve(Vertx vertx) {
+        HttpServer server = vertx.createHttpServer();
+        Router router = Router.router(vertx);
+
+        router.route()
+                .handler(BodyHandler.create())
+                .failureHandler(this::handleFailure);
+
+        router.route(HttpMethod.GET, "/catalog/model/:kind/:name")
+                .produces(MIME_TYPE_JSON)
+                .blockingHandler(this::handleCatalogModel);
+
+        router.route(HttpMethod.GET, "/catalog/capability/:name")
+                .produces(MIME_TYPE_JSON)
+                .blockingHandler(this::handleCatalogCapability);
+
+        router.route(HttpMethod.POST, "/inspect/:location")
+                .produces(MIME_TYPE_JSON)
+                .blockingHandler(this::handleInspect);
+
+        return server.requestHandler(router).listen(port, host).toCompletionStage();
+    }
+
+    private void handleFailure(RoutingContext ctx) {
+        LOGGER.info("", ctx.failure());
+        ctx.response()
+                .setStatusCode(400)
+                .setStatusMessage(
+                        ctx.failure().getCause() != null
+                                ? ctx.failure().getCause().getMessage()
+                                : ctx.failure().getMessage())
+                .end();
+    }
+
+    private void handleCatalogCapability(RoutingContext ctx) {
+        try {
+            final CamelCatalog catalog = loadCatalog(runtimeType, runtimeVersion);
+            final String name = ctx.pathParam("name");
+            final Optional<EntityRef> ref = catalog.findCapabilityRef(name);
+
+            if (ref.isPresent()) {
+                ctx.response()
+                        .putHeader(CONTENT_TYPE_HEADER, MIME_TYPE_JSON)
+                        .end(MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(ref.get()));
+            } else {
+                ctx.response()
+                        .setStatusCode(404)
+                        .end();
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void handleCatalogModel(RoutingContext ctx) {
+        try {
+            final CamelCatalog catalog = loadCatalog(runtimeType, runtimeVersion);
+            final String kind = ctx.pathParam("kind");
+            final String name = ctx.pathParam("name");
+            final BaseModel<?> model = catalog.model(Kind.valueOf(kind), name);
+
+            if (model != null) {
+                ctx.response()
+                        .putHeader(CONTENT_TYPE_HEADER, MIME_TYPE_JSON)
+                        .end(MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(model));
+            } else {
+                ctx.response()
+                        .setStatusCode(404)
```

Comment:
same as before, but I can see you refer to an url that does not exists, but imho empty response is also correct, could be an empty json response that would be better